e2e/instances/list/index.spec.ts

```diff
@@ -0,0 +1,388 @@
+    "there is no pagination when there aren't many instances"
+  ).not.toBeVisible();
+});
+
+test("test the pagination", async ({ page }) => {
+  test.setTimeout(200000);
+
+  await createWorkflow({
+    payload: childWorkflowContent,
+    urlParams: {
+      baseUrl: process.env.VITE_DEV_API_DOMAIN,
+      namespace,
+      name: "child.yaml",
+    },
+  });
+
+  await createWorkflow({
+    payload: parentWorkflowContent,
+    urlParams: {
+      baseUrl: process.env.VITE_DEV_API_DOMAIN,
+      namespace,
+      name: "parent.yaml",
+    },
+  });
+
+  const parentInstance = await runWorkflow({
+    urlParams: {
+      baseUrl: process.env.VITE_DEV_API_DOMAIN,
+      namespace,
+      path: "parent.yaml",
+    },
+  });
+
+  // there needs to be a discussion to check the total count
+  // expect(instancesList.instances.pageInfo.total, "length of the instance list should be 5").toBe(5);
+  await page.goto(`${namespace}/instances/`, { waitUntil: "networkidle" });
+
+  await expect(
+    page.getByTestId("pagination-wrapper"),
+    "there should be pagination component"
+  ).toBeVisible();
+
+  const btnPrev = page.getByTestId("pagination-btn-left");
+  const btnNext = page.getByTestId("pagination-btn-right");
+
+  //loop through all pages by clicking Next button
+  const testPages = [
+    Math.floor(Math.random() * 12),
+    Math.floor(Math.random() * 12),
+    Math.floor(Math.random() * 12),
+  ];
+
+  for (let p = 1; p < Math.ceil(170 / 15) + 1; p++) {
+    const activeBtn = page.getByTestId(`pagination-btn-page-${p}`);
+    await expect(
+      activeBtn,
+      "active button with the page number should have active class"
+    ).toHaveClass(/z-10 bg-gray-3 dark:bg-gray-dark-3/);
+
+    if (p === 1) {
+      await expect(
+        btnPrev,
+        "prev button should be disabled at page 1"
+      ).toBeDisabled();
+    } else if (p === Math.ceil(170 / 15)) {
+      await expect(
+        btnNext,
+        "next button should be disabled at last page"
+      ).toBeDisabled();
+    } else {
+      await expect(
+        btnNext,
+        "next button should be enabled except at page 1"
+      ).toBeEnabled();
+      await expect(
+        btnPrev,
+        "next button should be enabled except at last page"
+      ).toBeEnabled();
+    }
+
+    if (testPages.includes(p)) {
+      //test in the 3 pages the api response and ui
+      //pick the first element and compare with api response
+      const instancesListOfPage = await getInstances({
+        urlParams: {
+          baseUrl: process.env.VITE_DEV_API_DOMAIN,
+          namespace,
+          limit: 15,
+          offset: (p - 1) * 15,
+        },
+      });
+      const firstInstance = instancesListOfPage.instances.results[0];
+      const instanceItemId = page.getByTestId(
+        `instance-row-id-${firstInstance?.id}`
+      );
+      await expect(instanceItemId, "ItemId should have the id").toContainText(
+        firstInstance?.id.slice(0, 8) || ""
```

Comment:
I think this is an invalid test too. There is a scenario where you allow the instance id to be empty and the test to be valid. I think I understand why you are doing this, because `firstInstance.id.slice(0, 8)` (without the `?`) result in a typescript error, right? You can get around this by doing this

```
const firstInstance = instancesListOfPage.instances.results[0];
if (!firstInstance)
        throw new Error("there should be at least one instance");
```

this will make the test fail very early when the `firstInstance` is undefined and after this block, TypeScript will know that firstInstance can not be undefined at this point